packages/server/lib/controllers/auth/postBill.ts

```diff
@@ -257,7 +259,7 @@ export const postPublicBillAuthorization = asyncWrapper<PostPublicBillAuthorizat
             metadata: { providerConfigKey, connectionId }
         });
 
-        metrics.increment(metrics.Types.AUTH_FAILURE, 1, { auth_mode: 'BILL' });
+        metrics.increment(metrics.Types.AUTH_FAILURE, 1, { auth_mode: 'BILL', ...(config ? { provider: config.provider } : {}) });
```

Comment:
you could have set `config?.provider` for simplicity no?

Reply:
just checked. `dogstatsd` and our `metrics` abstraction don't accept undefined dimensions. type is `Record<string,  string | number>`